traffic_portal/app/src/traffic_portal_properties.json

```diff
@@ -201,6 +201,16 @@
           "url": "http://trafficcontrol.apache.org/"
         }
       ]
+    },
+    "oAuth": {
+      "_comment": "Opt-in OAuth properties for SSO login. See http://traffic-control-cdn.readthedocs.io/en/release-4.0.0/admin/quick_howto/oauth_login.html for more details. redirectUriParameterOverride defaults to redirect_uri if left blank.",
+      "enabled": false,
+      "oAuthUrl": "https://oauthProvider.example.com/auth",
+      "oAuthTokenQueryParam": "example_token_key",
+      "redirectUriParameterOverride": "example_redirect_url_key",
+      "clientId": "exampleClient",
+      "oAuthCodeTokenUrl": "https://oauthProvider.example.com/auth/token",
+      "clientSecret": "AbCd123"
```

Comment:
@jhg03a brought up a good point to me regarding storing secrets in this file. probably not a good idea as this file is "publicly" accessible - you might want to look into adding this secret to https://github.com/apache/trafficcontrol/blob/master/traffic_portal/conf/config.js (which is not publicly accessible) which gets read by server.js when  the server starts up.

Reply:
I actually think that only TO needs to know the client_secret, so you could just turn `whitelisted_oauth_urls` from an array into a map of the URLs to their associated client_secret. Then some parsing logic would need to change, and just modify how it's accessed in the route handler and you're done.

Reply:
i like the idea of moving the client secret into TO but the whitelisted urls are referencing the public key url so not really related to the client id.

Reply:
moved client secret to TO config